p8e-api/src/main/kotlin/io/provenance/engine/event/IndexHandler.kt

```diff
@@ -114,6 +123,36 @@ class IndexHandler(
                                     envelope.data.result.contract.recitalsList.map { it.signer.encryptionPublicKey.toPublicKey().toSha512Hex() }
                                 ), RequestOptions.DEFAULT
                             )
+                            // If the env is the invoker, create the data access message and put into a job.
+                            if(envelope.isInvoker == true) {
+                                val envelopeDataAccess = envelope.data.input.affiliateSharesList.map {
+                                    affiliateService.getAddress(
+                                        it.toPublicKey(), chaincodeProperties.mainNet
+                                    )
+                                }
+                                val existingScopeDataAccess = provenanceGrpcService.retrieveScopeData(envelope.data.input.scope.uuid.value).scope.scope.dataAccessList
+                                // Only perform job if data access will be updated
+                                if (envelopeDataAccess.any { it !in existingScopeDataAccess }) {
+                                    val scopeDataAccessRequest = p8e.Jobs.MsgAddScopeDataAccessRequest.newBuilder()
+                                        .addAllDataAccess(envelopeDataAccess)
+                                        .addAllSigners(envelope.data.result.signaturesList.map {
+                                            it.signer.signingPublicKey.toPublicKey().let {
+                                                affiliateService.getAddress(it, chaincodeProperties.mainNet)
+                                            }
+                                        })
+                                        .setScopeId(
+                                            envelope.data.input.ref.scopeUuid.value.toUuidProv().toAddress(
+                                                PROV_METADATA_PREFIX_SCOPE_ADDR
+                                            ).toByteString()
+                                        )
+                                        .setPublicKey(envelope.data.input.contract.invoker.encryptionPublicKey)
+                                        .build().takeIf { envelope.data.input.affiliateSharesList.isNotEmpty() }
```

Comment:
I think it would be more idiomatic kotlin to remote the `if` below and instead add

```kotlin
?.let { dataAccessService.addDataAccess(it) }
```